src/aktualizr_info/main.cc

```diff
@@ -91,6 +99,18 @@ int main(int argc, char **argv) {
       }
     }
 
+    std::string director_root;
+    bool has_metadata = storage->loadLatestRoot(&director_root, Uptane::RepositoryType::Director());
+    auto registered = storage->loadEcuRegistered();
+
+    while (wait_provisioning && !provisioned) {
+      registered = storage->loadEcuRegistered();
+      has_metadata = storage->loadLatestRoot(&director_root, Uptane::RepositoryType::Director());
+      provisioned = (registered && has_metadata);
```

Comment:
I'm not sure this `provisioned` variable is doing much anymore. Also, currently `storage->loadEcuRegistered()` and `storage->loadLatestRoot()` will always run at least twice. If the while loop's condition directly used `has_metadata` and `registered`, you wouldn't have that problem.

Reply:
Sure, thanks!

Reply:
> Seems cool, looks quite close! Thanks.
> 
> @rdanitz @balajirrao @Raigi Is this useful for your needs?

Currently I have a loop that will check aktualizr-info multiple times for provisioned: yes and metadata: yes if it received false. But it might be faster and better with aktualizr-info-wait-until-provisioned when I am creating like 50 devices.

Reply:
Cool, that's exactly the type of thing we're hoping this will help optimize.